test/libsolidity/syntaxTests/literalSuffixes/parameters/too_many_parameters.sol

```diff
@@ -24,9 +24,9 @@ contract C {
 // ----
 // TypeError 9128: (27-45): Functions that take 3 or more arguments cannot be used as literal suffixes.
 // TypeError 9128: (120-165): Functions that take 3 or more arguments cannot be used as literal suffixes.
-// TypeError 8838: (502-520): The type of the literal cannot be converted to the parameters of the suffix function.
-// TypeError 8838: (530-548): The type of the literal cannot be converted to the parameters of the suffix function.
-// TypeError 8838: (629-649): The type of the literal cannot be converted to the parameters of the suffix function.
-// TypeError 8838: (659-679): The type of the literal cannot be converted to the parameters of the suffix function.
+// TypeError 8838: (502-520): This fractional number cannot be decomposed into a mantissa and decimal exponent that fit the range of parameters of the suffix function. The exponent is out of range of type string.
+// TypeError 8838: (530-548): This fractional number cannot be decomposed into a mantissa and decimal exponent that fit the range of parameters of the suffix function. The mantissa is out of range of type string.
+// TypeError 8838: (629-649): This fractional number cannot be decomposed into a mantissa and decimal exponent that fit the range of parameters of the suffix function. The exponent is out of range of type string.
+// TypeError 8838: (659-679): This fractional number cannot be decomposed into a mantissa and decimal exponent that fit the range of parameters of the suffix function. The mantissa is out of range of type string.
```

Comment:
This sounds weird :) We should only say this when the type of the parameter is a number.

Actually, this should really be caught even earlier, when type-checking the function definition. Having a suffix function that accepts two arguments where both aren't integers should be an error. We could not do that back when all functions could be used as suffixes, but now that we have the `suffix` modifier, we should move the check.

Reply:
I have added the check in `visit(FunctionDefinition)` which then raises an [error](https://github.com/ethereum/solidity/blob/a1412bac2f3d1ddb43e1d78864673b3fd553853a/libsolidity/analysis/TypeChecker.cpp#L451-L466) if any or both the paramters are not integer. 
I also kept this case being checked and reported in `endVisit(Literal)` with same error id `8838` and a different [error message](https://github.com/ethereum/solidity/blob/a1412bac2f3d1ddb43e1d78864673b3fd553853a/libsolidity/analysis/TypeChecker.cpp#L4093-L4095).